### PR TITLE
Fixed many links across all applicable files

### PR DESCRIPTION
**Pull request by zneix**
Most importantly, updated gh pages domain to communi.github.io, as github.com subdomains were looooong deprecated and recently stopped working.
Updated all links to https, because it's not 2013 anymore.
Updated all the links on ircv3.net that returned `301` and further redirects.

### CODE REVIEW

**Comment by zneix** [comment]
Also, repository name could use a name change to reflect the move from `communit.github.com` to `communi.github.io` however I leave this up to maintainers :)

**Review by jpnurmi — APPROVED** [review]
Thanks, LGTM :+1:

**Pull request merged**
